### PR TITLE
CheckBox: add different iconSizeRatio value for different themes and add getComputedStyles method to compute size

### PR DESCRIPTION
**Pull request by Zedwag**

### CODE REVIEW

**Comment by ksercs** [comment]
cannot we get rid of excess css rules for grid checkbox?

File: js/renovation/ui/editors/check_box.tsx [hunk 1]
```diff
@@ -161,10 +161,16 @@ export class CheckBox extends JSXComponent(CheckBoxProps) {
     const { iconWidth, iconHeight } = this.props;
 
     if (iconElement !== null && iconElement !== undefined) {
+      const isCompactTheme = current() ? current().includes('compact') : false;
```
Comment (ksercs) [inline js/renovation/ui/editors/check_box.tsx]:
```suggestion
      const isCompactTheme = current()?.includes('compact');
```

File: js/renovation/ui/editors/check_box.tsx [hunk 2]
```diff
@@ -161,10 +161,16 @@ export class CheckBox extends JSXComponent(CheckBoxProps) {
...
+      const defaultFontSize = isCompactTheme ? 12 : 16;
+      const isMaterialTheme = isMaterial(current());
+      let defaultIconSize = isMaterialTheme ? 18 : 22;
+      if (isCompactTheme) {
+        defaultIconSize = 16;
+      }
```
Comment (ksercs) [inline js/renovation/ui/editors/check_box.tsx:170]: isn't it better to use switch here?

File: js/renovation/ui/editors/__tests__/check_box.test.tsx [hunk 3]
```diff
@@ -266,19 +266,21 @@ describe('CheckBox', () => {
...
-        it("should set default font size if icon element's 'offsetHeight'/'offsetWidth' fields are not defined", () => {
-          const checkBox = new CheckBox({ iconHeight: '22px', iconWidth: '22px' });
-          checkBox.iconRef = React.createRef() as any;
-          checkBox.iconRef.current = {
-            offsetHeight: undefined,
-            offsetWidth: undefined,
-            style: {},
-          } as any;
-          checkBox.updateIconFontSize();
+        each(['standart', 'compact', undefined])
```
Comment (ksercs) [inline js/renovation/ui/editors/__tests__/check_box.test.tsx]: `theme=undefined` looks strange. Can we write a separate test to describe the case more specific?

File: js/renovation/ui/editors/check_box.tsx (hunk 2 above)
Comment (Zedwag) [reply js/renovation/ui/editors/check_box.tsx:170]: nope, look weird
image

File: js/renovation/ui/editors/__tests__/check_box.test.tsx (hunk 3 above)
Comment (Zedwag) [reply js/renovation/ui/editors/__tests__/check_box.test.tsx]: we can, but new test will almost completely duplicate these

File: js/renovation/ui/editors/check_box.tsx (hunk 1 above)
Comment (Zedwag) [reply js/renovation/ui/editors/check_box.tsx]: done

File: js/renovation/ui/editors/check_box.tsx [hunk 4]
```diff
@@ -160,13 +160,23 @@ export class CheckBox extends JSXComponent(CheckBoxProps) {
...
       const iconFontSizeRatio = defaultFontSize / defaultIconSize;
+
+      const width = typeof iconWidth === 'number'
+        ? iconWidth
+        : parseInt(window.getComputedStyle(iconElement).width, 10);
+      const height = typeof iconHeight === 'number'
+        ? iconHeight
+        : parseInt(window.getComputedStyle(iconElement).height, 10);
```
Comment (ksercs) [inline js/renovation/ui/editors/check_box.tsx]: let's call `getComputedStyle` only once

Comment (Zedwag) [reply js/renovation/ui/editors/check_box.tsx]: yeah, right

**Review by ksercs — APPROVED** [review]

**Review by vconst — APPROVED** [review]

**Review by DokaRus — APPROVED** [review]

**Review by alexanderPolosatov — APPROVED** [review]

**Pull request merged**